include/transmem/transmem.h

```diff
@@ -59,41 +60,64 @@ class EmptyLinkQueryException : public std::runtime_error {
     Link emptyLink;
 };
 
-
-/****************************
- * PATH                     *
- ****************************/
+/********
+ * PATH *
+ ********/
 
 struct Path {
+
     FrameID src;
     FrameID dst;
     std::vector< std::reference_wrapper<Link> > links;
 
     void writeJSON(QJsonObject &json) const;
 };
 
+/************************************
+ * STAMPED AND RATED TRANSFORAMTION *
+ * **********************************/
+
+/* REMARK:
+   New return type of transmem. Encodes not just the transformation but also
+   some information about the quality. Unnecessary convertion between matrix and quaternion
+   are no longer needed and the return type is much more flexible, one can add additional
+   stuff without changing interface to much. */
+
+struct StampedAndRatedTransformation {
+
+    // Quaternions encoding the transformation.
+    QQuaternion qRot;
+    QQuaternion qTra{0,0,0,0};
```

Comment:
It is very confusing for the reading when a translation is encoded in a quaternion. Even though there might be some contexts in which this makes sense, in common context it creates confusion. Please change indeed.